mobile/native/src/max_quantity.rs

```diff
@@ -123,14 +189,50 @@ fn calculate_max_quantity(
         leverage,
     );
 
-    Decimal::try_from(max_quantity.floor()).expect("to fit into decimal")
+    Decimal::try_from((max_quantity + open_quantity).floor()).expect("to fit")
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
     use rust_decimal_macros::dec;
 
+    #[test]
+    fn test_calculate_max_quantity_with_open_quantity() {
+        let price = Decimal::new(22001, 0);
+
+        let max_coordinator_margin = Amount::from_sat(765_763);
+        let max_trader_margin = Amount::from_sat(747_499);
+
+        let trader_leverage = 2.0;
+        let coordinator_leverage = 2.0;
+        let order_matching_fee_rate = dec!(0.003);
+        let open_quantity = dec!(323);
+
+        let max_quantity = calculate_max_quantity(
+            price,
+            max_coordinator_margin,
+            max_trader_margin,
+            None,
+            coordinator_leverage,
+            trader_leverage,
+            order_matching_fee_rate,
+            Amount::from_sat(4459),
+            open_quantity,
+        );
+
+        assert_eq!(dec!(648), max_quantity);
+
+        // Ensure that the coordinator has enough funds for the trade
+        let coordinator_margin = calculations::calculate_margin(
+            price.to_f32().unwrap(),
+            (max_quantity - open_quantity).to_f32().unwrap(),
+            coordinator_leverage,
+        );
+
+        assert!(Amount::from_sat(coordinator_margin) < max_coordinator_margin);
```

Comment:
🙃
no need to convert it to amount which will internally take the u64 anyways 

```suggestion
assert!(coordinator_margin < max_coordinator_margin.to_sat());
```